app/controllers/mentee_profiles_controller.rb

```diff
@@ -1,6 +1,24 @@
 class MenteeProfilesController < ApplicationController
-	def show
-	end
-	def dashboard
-	end
+  def show
+    @user = User.find_by(id: params[:id])
+    @mentor = User.find_by(id: @user.matched_id)
+
+  end
+  def dashboard
+    @mentor = User.find_by(id: current_user.matched_id)
+  end
+  def missing_mentor
+    @user = current_user
+    if !@user.blank? 
```

Comment:
the whole controller should have:
`before_action :require_login` - checks if user is logged in
also for all methods except show we should check if logged user is mentee (so another before_action). As for show
we should check whether logged user is mentee or organiser and only then allowed to proceed else redirect to home page with flash "You're not authorized to access this page" or something like that